rotkehlchen/tests/api/test_bitcoin_transactions.py

```diff
@@ -125,3 +152,69 @@ def test_query_transactions(
         assert assert_proper_response_with_result(response, rotkehlchen_api_server, async_query)
 
     assert mock_decode_tx.call_count == 0
+
+
+@pytest.mark.vcr
+@pytest.mark.freeze_time('2025-07-14 08:00:00 GMT')
+@pytest.mark.parametrize('ethereum_accounts', [[]])
+@pytest.mark.parametrize('bch_accounts', [['bitcoincash:qpplh0vyfn67cupcmhq4g2dt3s50rlarmclu9vnndt']])  # noqa: E501
+@pytest.mark.parametrize('legacy_messages_via_websockets', [True])
+def test_query_bch_transactions(
+        rotkehlchen_api_server: 'APIServer',
+        bch_accounts: list['BTCAddress'],
+        websocket_connection: WebsocketReader,
+) -> None:
+    """Test that bitcoin cash transactions are properly queried via the api."""
+    async_query = random.choice([False, True])
+    for json, expected_len in (
+        ({'async_query': async_query, 'to_timestamp': 1700000000}, 4),  # query partial range first
+        ({'async_query': async_query}, 7),  # then query the rest
+    ):
+        events = do_tx_query_and_get_events(
+            rotkehlchen_api_server=rotkehlchen_api_server,
+            websocket_connection=websocket_connection,
+            accounts=bch_accounts,
+            async_query=async_query,
+            json=json,
+            expected_len=expected_len,
+            chain=SupportedBlockchain.BITCOIN_CASH,
+        )
+
+    assert events[5:] == [HistoryEvent(
+        identifier=5,
+        event_identifier=(event_identifier := f'{BCH_EVENT_IDENTIFIER_PREFIX}3944ec023a1a4004d26c476051160ab97c1004a5a34799fc197c885acc745ead'),  # noqa: E501
+        sequence_index=0,
+        timestamp=(timestamp := TimestampMS(1749190001000)),
+        location=Location.BITCOIN_CASH,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=A_BCH,
+        amount=FVal(fee_amount := '0.00007592'),
+        location_label=(user_address := bch_accounts[0]),
+        notes=f'Spend {fee_amount} BCH for fees',
+    ), HistoryEvent(
+        identifier=6,
+        event_identifier=event_identifier,
+        sequence_index=1,
+        timestamp=timestamp,
+        location=Location.BITCOIN_CASH,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.NONE,
+        asset=A_BCH,
+        amount=FVal(spend_amount := '0.03559966'),
+        location_label=user_address,
+        notes=f'Send {spend_amount} BCH to bitcoincash:qr40efj25rw7lmr4qr90636xne2gsmq39ugdsw5k3g',
```

Comment:
a question. do you check if the receiving address is tracked(whether legacy or cashaddr format)?